### PR TITLE
Buff/dot improvements

### PR DESCRIPTION
**Pull request by xpdota**
- Fixes a bug that causes ability/status icons to not load sometimes
- Add extra description parts and description overrides to buffs
- Add 'indefinite' buff and dot duration
- Added support for buff stacks, but this is only implemented for icons

### CODE REVIEW

**Review by xiashtra — CHANGES_REQUESTED** [review]
Attempting to have a gcd action that applies an indefinite duration buff causes the buff to not be applied at all.

**Comment by xiashtra** [comment]
Infinite duration buffs do not work at all with this PR.

File: packages/frontend/src/scripts/sims/blu/blu_common.ts [hunk 1]
```diff
@@ -80,35 +80,35 @@ const Boost: Buff = {
...
+const WaxingNocturne = {
     name: "Waxing Nocturne",
     duration: 15,
     selfOnly: true,
     effects: {
         dmgIncrease: 0.50
     },
     statusId: 1718
-}
+} as const satisfies Buff; // This allows the 'indefinite' check to be skipped
```
[the remark this answers is not in the document]
Comment (xpdota) [reply packages/frontend/src/scripts/sims/blu/blu_common.ts:91]: Hmm, I think these end up not being needed. I needed them at one point to make it compile without having additional checks within the BLU-specific sim code.

**Comment by xiashtra** [comment]
> Should all be addressed now

Working as expected now.

**Pull request merged**